oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -126,6 +126,138 @@ static ucp_request_param_t mca_spml_ucx_request_param_b = {
 };
 #endif
 
+unsigned
+mca_spml_ucx_mem_map_flags_symmetric_rkey(struct mca_spml_ucx *spml_ucx)
+{
+#if HAVE_DECL_UCP_MEM_MAP_SYMMETRIC_RKEY
+    if (spml_ucx->symmetric_rkey) {
+        return UCP_MEM_MAP_SYMMETRIC_RKEY;
+    }
+#else
+    return 0;
+#endif
+}
+
+int mca_spml_ucx_rkey_store_create(mca_spml_ucx_rkey_store_t *store,
+                                   ucp_context_h ucp_context,
+                                   int size)
+{
+    store->array = calloc(size, sizeof(*store->array));
+    if (store->array == NULL) {
+        return OSHMEM_ERR_OUT_OF_RESOURCE;
+    }
+
+    store->size        = size;
+    store->count       = 0;
+    store->ucp_context = ucp_context;
+    return OSHMEM_SUCCESS;
+}
+
+void mca_spml_ucx_rkey_store_destroy(mca_spml_ucx_rkey_store_t *store)
+{
+    int i;
+
+    for (i = 0; i < store->count; i++) {
+        assert(store->array[i].refcnt == 0);
+        ucp_rkey_destroy(store->array[i].rkey);
+    }
+
+    free(store->array);
+    store->count = 0;
+    store->size  = 0;
+}
+
+static int mca_spml_ucx_rkey_store_find(const mca_spml_ucx_rkey_store_t *store,
```

Comment:
need to double check rkey and its worker

Reply:
added worker with storage and as parameter for store get, checking it as it could be different with ucx `async_progress` parameter set. store put path can rely on pointer value.